test/extended/util/cli.go

```diff
@@ -148,6 +150,17 @@ func (c *CLI) SetupProject(name string, kubeClient *kclient.Client, _ map[string
 		e2e.Logf("Failed to create a project and namespace %q: %v", c.Namespace(), err)
 		return nil, err
 	}
+	if err := wait.ExponentialBackoff(kclient.DefaultBackoff, func() (bool, error) {
```

Comment:
We should have a helper to create a SAR condition "CanI"

On Thu, Apr 21, 2016 at 6:26 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> I'll need to cut and paste because integration package loads etcd.  Also,
> I don't want to poll, I want to backoff.
>
> On Thu, Apr 21, 2016 at 3:50 PM, David Eads <notifications@github.com>
> wrote:
>
>> In test/extended/util/cli.go
>> <https://github.com/openshift/origin/pull/8585#discussion_r60644156>:
>>
>> > @@ -148,6 +150,17 @@ func (c *CLI) SetupProject(name string, kubeClient *kclient.Client, _ map[string
>> >  		e2e.Logf("Failed to create a project and namespace %q: %v", c.Namespace(), err)
>> >  		return nil, err
>> >  	}
>> > +	if err := wait.ExponentialBackoff(kclient.DefaultBackoff, func() (bool, error) {
>>
>> Ick.
>> https://github.com/openshift/origin/blob/master/test/util/policy.go#L20.
>> Even a copy/paste would be preferable.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/8585/files/577972ccdb7de74c7cee66c1aff7f57050fc4f5d#r60644156>
>>
>
>